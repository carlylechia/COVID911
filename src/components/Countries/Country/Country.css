#country-card {
  display: grid;
  grid-template-columns: 30% 70%;
  grid-template-rows: 50% 1fr 1fr;
}

#country-map {
  justify-self: center;
}

#country-name {
  align-self: end;
}

#new-cases {
  grid-row: 2/2;
  grid-column: 2/2;
}

#total {
  grid-row: 3/3;
  grid-column: 2/2;
}
