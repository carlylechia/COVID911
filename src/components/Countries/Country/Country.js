import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Country = ({
  name, image, total, newCases, pageUrl,
}) => (
  <NavLink
    to={`${pageUrl}`}
    className="country-card grow pointer flex flex-column justify-center items-center"
    style={{ minHeight: '25vh' }}
  >
    <div
      style={{
        width: '10rem',
        height: '60%',
        background: ` url(${image}) no-repeat 50% 50%`,
        backgroundSize: 'auto 80%',
        opacity: '0.2',
      }}
      loading="lazy"
    />
    <p className="pr3 white f4 b" style={{ fontSize: '1.3rem', color: 'aliceblue', textAlign: 'center' }}>{name}</p>
    <p className="db pr3 white f5 b" style={{ fontSize: '1rem', color: 'azure', textAlign: 'center' }}>
      Total:
      {` ${total}`}
    </p>
    <p className="db pr3 white f5 b" style={{ fontSize: '1rem', color: 'azure', textAlign: 'center' }}>
      New Cases:
      {` ${newCases}`}
    </p>
  </NavLink>
);
Country.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  total: PropTypes.number.isRequired,
  newCases: PropTypes.number.isRequired,
  pageUrl: PropTypes.string.isRequired,
};

export default Country;
