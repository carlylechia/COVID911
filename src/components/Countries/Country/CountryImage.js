import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

export const getCountrySlug = (name) => {
  let countryName = name.replace(/[_]/g, '-').replace('*', '').replace(',', '');
  if (countryName === 'diamond-princess' || countryName === 'ms-zaandam') countryName = 'ship';
  if (countryName === 'south-sudan') countryName = 'sudan';
  if (countryName.includes('congo')) countryName = 'congo';
  if (countryName === 'west-bank-and-gaza') countryName = 'palestine';
  if (countryName.includes('timor')) countryName = 'timor';
  if (countryName === 'us') countryName = 'usa';

  return countryName;
};

export const useCountryImage = (src) => {
  const [url, setUrl] = useState();

  useEffect(() => {
    if (!url) return;
    const img = new Image();
    img.onerror = () => {
      setUrl(`https://mapsvg.com/static/maps/geo-calibrated/${getCountrySlug(src)}.svg`);
    };
    img.src = url;
  }, [url]);

  useEffect(() => {
    setUrl(src ? `/images/${getCountrySlug(src)}.png` : '');
  }, [src]);

  return url;
};

const CountryImage = ({ src, alt, className }) => {
  const url = useCountryImage(src);
  return <img src={url} alt={alt} className={className} />;
};

CountryImage.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
};

CountryImage.defaultProps = {
  className: '',
};

export default CountryImage;
