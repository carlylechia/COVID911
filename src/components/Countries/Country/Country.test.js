import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../Redux/createStore';
import Country from './Country';

describe('Country component test', () => {
  test('Country snapshot', () => {
    const props = {
      name: 'country 1',
      image: 'country 1 image',
      total: 100,
      newCases: 50,
      pageUrl: 'url',
    };
    const {
      name, image, total, newCases, pageUrl,
    } = props;
    const country = render(
      <Router>
        <Provider store={store}>
          <Country
            name={name}
            image={image}
            total={total}
            newCases={newCases}
            pageUrl={pageUrl}
          />
        </Provider>
      </Router>,
    );
    expect(country).toMatchSnapshot();
  });
});
