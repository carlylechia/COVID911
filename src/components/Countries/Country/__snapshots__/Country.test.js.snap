// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component test Country snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="country-card grow pointer flex flex-column justify-center items-center"
        href="/url"
        style="min-height: 25vh;"
      >
        <div
          loading="lazy"
          style="width: 10rem; height: 70%; background-size: auto 80%; opacity: 0.2;"
        />
        <p
          class="pr3 white f4 b"
          style="font-size: 1.3rem; color: aliceblue;"
        >
          country 1
        </p>
        <span
          class="db pr3 white f5 b"
          style="font-size: 1rem; color: azure;"
        >
          Total:
           100
        </span>
        <p
          class="db pr3 white f5 b"
          style="font-size: 1rem; color: azure;"
        >
          New Cases:
           50
        </p>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="country-card grow pointer flex flex-column justify-center items-center"
      href="/url"
      style="min-height: 25vh;"
    >
      <div
        loading="lazy"
        style="width: 10rem; height: 70%; background-size: auto 80%; opacity: 0.2;"
      />
      <p
        class="pr3 white f4 b"
        style="font-size: 1.3rem; color: aliceblue;"
      >
        country 1
      </p>
      <span
        class="db pr3 white f5 b"
        style="font-size: 1rem; color: azure;"
      >
        Total:
         100
      </span>
      <p
        class="db pr3 white f5 b"
        style="font-size: 1rem; color: azure;"
      >
        New Cases:
         50
      </p>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
